spec/workers/subscriber_message_delivery_worker_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'spec_helper'
+
+describe SubscriberMessageDeliveryWorker do
+  let(:delivery) { Fabricate(:delivery) }
+  it 'sends email' do
+    subscriber = delivery.subscriber
+    message = delivery.message
+
+    subject.perform(delivery.id)
+    message_delivery = ActionMailer::Base.deliveries.last
+    expect(message_delivery.to).to eql([subscriber.email])
+
+    msg = message_delivery.to_s
+    expect(msg).to include(message.html_body)
+    expect(msg).to include(message.text_body)
+  end
+
+  it 'sets delivered_at' do
+    subject.perform(delivery.id)
+    expect(delivery.reload.delivered_at).to be_present
```

Comment:
maybe a nitpick but couldn't you freeze time with timecop and check if the delivered_at matches it?